### PR TITLE
fix(Android): fixed deprecated APPLICATION_ID for failed builds

### LINKED ISSUE #3254: cannot find symbol BuildConfig.APPLICATION_ID
## Issue

Describe your issue here

```
node_modules@react-native-firebase\app\android\src\reactnative\java\io\invertase\firebase\app\ReactNativeFirebaseAppInitProvider.java:25: error: cannot find symbol
    BuildConfig.APPLICATION_ID + ".reactnativefirebaseappinitprovider";
               ^
  symbol:   variable APPLICATION_ID
  location: class BuildConfig
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':@react-native-firebase_app:compileDebugJavaWithJavac'.
```

---

### Javascript

<details><summary>Click To Expand</summary>
<p>

#### `package.json`:

```json
# N/A
{
  "name": "appname_mobile",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "test": "jest",
    "lint": "eslint .",
    "postinstall": "npx rn-nodeify --install \"fs, util, path, tls, stream,buffer,global,process\" --hack  \"node ./android-release-gradle-fix.js\"",
    "build:ios": "react-native bundle --entry-file='index.js' --bundle-output='./ios/main.jsbundle' --dev=false --platform='ios'"
  },
  "dependencies": {
    "@react-native-community/async-storage": "1.7.1",
    "@react-native-firebase/analytics": "^6.3.4",
    "@react-native-firebase/app": "^6.3.4",
    "@react-native-firebase/crashlytics": "^6.3.4",
    "asyncstorage-down": "^4.2.0",
    "aws-iot-device-sdk": "^2.2.1",
    "aws-sdk": "^2.601.0",
    "axios": "0.19.0",
    "buffer": "^4.9.1",
    "events": "^3.1.0",
    "native-base": "^2.13.8",
    "path-browserify": "0.0.0",
    "process": "^0.11.10",
    "react": "16.9.0",
    "react-native": "0.61.5",
    "react-native-appearance": "0.3.1",
    "react-native-check-box": "^2.1.7",
    "react-native-config": "^0.12.0",
    "react-native-confirmation-code-input": "1.0.4",
    "react-native-dashed-progress": "^1.0.1",
    "react-native-fs": "^2.16.6",
    "react-native-gesture-handler": "^1.5.2",
    "react-native-level-fs": "^3.0.1",
    "react-native-localization": "2.1.6",
    "react-native-logs": "^2.0.0",
    "react-native-modal": "^11.5.3",
    "react-native-paper": "^3.5.1",
    "react-native-reanimated": "^1.4.0",
    "react-native-slider": "^0.11.0",
    "react-native-splash-screen": "^3.2.0",
    "react-native-tab-view": "^2.11.0",
    "react-native-uuid": "^1.4.9",
    "react-native-vector-icons": "^6.6.0",
    "react-navigation": "4.0.10",
    "react-navigation-stack": "1.10.3",
    "react-navigation-tabs": "2.6.2",
    "react-redux": "^7.1.3",
    "readable-stream": "^1.0.33",
    "redux": "^4.0.5",
    "redux-thunk": "^2.3.0",
    "reselect": "^4.0.0",
    "rn-nodeify": "^10.2.0",
    "stream-browserify": "^1.0.0",
    "util": "^0.10.4"
  },
  "devDependencies": {
    "@babel/core": "^7.7.4",
    "@babel/runtime": "^7.7.4",
    "@react-native-community/eslint-config": "^0.0.5",
    "babel-jest": "^24.9.0",
    "eslint": "^6.7.2",
    "jest": "^24.9.0",
    "metro-react-native-babel-preset": "^0.57.0",
    "react-test-renderer": "16.9.0"
  },
  "jest": {
    "preset": "react-native"
  },
  "react-native": {
    "path": "path-browserify",
    "fs": "react-native-level-fs",
    "_stream_transform": "readable-stream/transform",
    "_stream_readable": "readable-stream/readable",
    "_stream_writable": "readable-stream/writable",
    "_stream_duplex": "readable-stream/duplex",
    "_stream_passthrough": "readable-stream/passthrough",
    "stream": "stream-browserify",
    "tls": false
  },
  "browser": {
    "path": "path-browserify",
    "fs": "react-native-level-fs",
    "_stream_transform": "readable-stream/transform",
    "_stream_readable": "readable-stream/readable",
    "_stream_writable": "readable-stream/writable",
    "_stream_duplex": "readable-stream/duplex",
    "_stream_passthrough": "readable-stream/passthrough",
    "stream": "stream-browserify",
    "tls": false
  }
}

```

#### `firebase.json` for react-native-firebase v6:

```json
# N/A
```

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
# N/A
```

#### `AppDelegate.m`:

```objc
// N/A
```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

- [x ] my application is an AndroidX application?
- [x ] I am using `android/gradle.settings` `jetifier=true` for Android compatibility?
- [ ] I am using the NPM package `jetifier` for react-native compatibility?

#### `android/build.gradle`:

```groovy
apply plugin: "com.android.application"
apply from: project(':react-native-config').projectDir.getPath() + "/dotenv.gradle"
apply from: "../../node_modules/react-native-vector-icons/fonts.gradle"
apply plugin: 'com.google.gms.google-services'
apply plugin: 'com.google.firebase.crashlytics'
.
.
.
dependencies {
    implementation project(':react-native-splash-screen')
    implementation project(path: ":@react-native-firebase_app")
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation "com.facebook.react:react-native:+"  // From node_modules

    implementation 'com.google.firebase:firebase-analytics:17.2.2'
    implementation 'com.google.firebase:firebase-crashlytics:17.0.0-beta01'


    if (enableHermes) {
        def hermesPath = "../../node_modules/hermes-engine/android/";
        debugImplementation files(hermesPath + "hermes-debug.aar")
        releaseImplementation files(hermesPath + "hermes-release.aar")
    } else {
        implementation jscFlavor
    }
}
```

#### `android/app/build.gradle`:

```groovy
buildscript {
    ext {
        buildToolsVersion = "28.0.3"
        minSdkVersion = 16
        compileSdkVersion = 28
        targetSdkVersion = 28
    }
    repositories {
        google()
        jcenter()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:4.0.0-alpha09'
        classpath 'com.google.gms:google-services:4.3.3'
        classpath 'com.google.firebase:firebase-crashlytics-gradle:2.0.0-beta02'

        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}

allprojects {
    repositories {
        mavenLocal()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url("$rootDir/../node_modules/react-native/android")
        }
        maven {
            // Android JSC is installed from npm
            url("$rootDir/../node_modules/jsc-android/dist")
        }

        google()
        jcenter()
        maven { url 'https://jitpack.io' }
    }
}
```

#### `android/settings.gradle`:

```groovy
// N/A
```

#### `MainApplication.java`:

```java
// N/A
```

#### `AndroidManifest.xml`:

```xml
<!-- N/A -->
```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
error Failed to install the app. Make sure you have the Android development environment set up: https://facebook.github.io/react-native/docs/getting-started.html#android-development-environment. Run CLI with --verbose flag for more details.
Error: Command failed: gradlew.bat app:installDebug -PreactNativeDevServerPort=8081
D:\<path to project>\node_modules\@react-native-firebase\app\android\src\reactnative\java\io\invertase\firebase\app\ReactNativeFirebaseAppInitProvider.java:25: error: cannot find symbol
    BuildConfig.APPLICATION_ID + ".reactnativefirebaseappinitprovider";
               ^
  symbol:   variable APPLICATION_ID
  location: class BuildConfig
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':@react-native-firebase_app:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.
```

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [x] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - `6.3.4`
- **`Firebase` module(s) you're using that has the issue:**
  - `e.g. Instance ID`
- **Are you using `TypeScript`?**
  - `N` & `VERSION`
  
</p>
</details>

---

Think `react-native-firebase` is great? Please consider supporting all of the project maintainers and contributors by donating via our [Open Collective](https://opencollective.com/react-native-firebase/donate) where all contributors can submit expenses. [[Learn More]](https://invertase.io/oss/react-native-firebase/contributing/donations-expenses)

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

### PR DESCRIPTION
**Pull request by FrenchBully**
### Description
This PR is to resolve the issue with deprecated APPLICATION_ID on Android documented [here](https://android.googlesource.com/platform/tools/base/+/51644d6f302ea8c3616d98afde0b4ee30d7d4741) 

Unable to build App to Android: 
```
node_modules@react-native-firebase\app\android\src\reactnative\java\io\invertase\firebase\app\ReactNativeFirebaseAppInitProvider.java:25: error: cannot find symbol
    BuildConfig.APPLICATION_ID + ".reactnativefirebaseappinitprovider";
               ^
  symbol:   variable APPLICATION_ID
  location: class BuildConfig
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':@react-native-firebase_app:compileDebugJavaWithJavac'.
```

### Related issues
Fixes #3254
[Issue - 3254: deprecated APPLICATION_ID](https://github.com/invertase/react-native-firebase/issues/3254)

### Release Summary
- fix for deprecated `APPLICATION_ID` to use `LIBRARY_PACKAGE_NAME`

### Checklist

- I read the [Contributor Guide](/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [x] Yes
- My change supports the following platforms;
  - [x] `Android`
  - [x] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [x] Yes
  - [ ] No

### Test Plan

Screen Shot 2020-05-28 at 1 47 38 PM

Screen Shot 2020-05-28 at 1 48 20 PM

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

**Comment by mikehardy** [comment]
The other way to fix this was to refactor and take BuildConfig out, vs using these statics, I'm not sure the LIBRARY_PACKAGE_NAME is the right direction? Is it a breaking change even, since it would be a different constant I think? 

From @Salakar  on linked issue

> Thanks @mikehardy for investigating.
> 
> We could probably just refactor this not to rely on BuildConfig, I think there's an underlying gradle change where sub projects no longer have application id generated into the build config.
> 
> We can switch this out for Context.getPackageName() in the base class, should solve the issue. PRs welcome otherwise we'll get to this as soon as possible.

**Comment by FrenchBully** [comment]
@mikehardy does the recent changes fit the intended use or are you expecting another way to fetch package name? Sorry, I forgot to mention I'm not a Java engineer. Just learning through it. Thanks for any feedback and insights.

**Comment by mikehardy** [comment]
@Salakar Android Studio 4.0 just came out today so everyone is going to break android builds unless either this is merged or you refactor out in the way you mentioned and I quoted above

**Comment by mikehardy** [comment]
@FrenchBully I think Mike (Salakar) was thinking of something different but we've run out of time since Android Studio 4 just went general release. We'll see what he says

**Comment by mikehardy** [comment]
Actually your last commit looks like what was sketched out (sorry I didn't see it until now) investigating why it didn't compile. Must be something simple - this isn't a crazy change and I am a Java programmer :-), give me a minute

File: packages/crashlytics/android/src/main/java/io/invertase/firebase/crashlytics/Constants.java [hunk 1]
```diff
@@ -1,4 +1,5 @@
 package io.invertase.firebase.crashlytics;
+import io.invertase.firebase.common.ReactNativeFirebaseInitProvider;
```
Comment (mikehardy) [inline packages/crashlytics/android/src/main/java/io/invertase/firebase/crashlytics/Constants.java:2]:
```suggestion
import io.invertase.firebase.app.ReactNativeFirebaseApp;
import io.invertase.firebase.common.ReactNativeFirebaseInitProvider;
```

**Comment by mikehardy** [comment]
..actually I had the ability to commit it, we'll see
(sorry for bumbling through this - I haven't gone through a react-native-firebase change myself in a while, I'm a bit rusty)

**Comment by mikehardy** [comment]
My attempt only made it worse ;-), @Salakar just fixed it for real and released - @FrenchBully thanks for giving it a go either way, definitely got it moving and just in time - much appreciated

**Pull request closed without merging**